### PR TITLE
Fix IPv6 zone handling, lowercase in domain name and compressed ip ad…

### PR DESCRIPTION
**Pull request by socketpair**
…dresses.

### CODE REVIEW

File: yarl/__init__.py [hunk 1]
```diff
@@ -160,16 +160,16 @@ def __init__(self, val='', *, encoded=False, strict=None):
...
-                except UnicodeEncodeError:
-                    netloc = netloc.encode('idna').decode('ascii')
                 else:
+                    (ip, *zone) = netloc.split('%', 1)
                     try:
-                        ip = ip_address(netloc)
+                        ip = ip_address(ip)
                     except ValueError:
-                        pass
+                        netloc = netloc.encode('idna').decode('ascii')
```
Comment (socketpair) [inline yarl/__init__.py]:
```
In [9]: timeit 'http://lasdhglajkdshflkajsdfh.asdf.asdf.asdf.asdf.asdf.rtyty'.encode('ascii')
1000000 loops, best of 3: 281 ns per loop

In [10]: timeit 'http://lasdhglajkdshflkajsdfh.asdf.asdf.asdf.asdf.asdf.rtyty'.encode('idna').decode('ascii')
100000 loops, best of 3: 2.79 µs per loop

```

Comment (socketpair) [reply yarl/__init__.py]: Is it the case ?

Comment (asvetlov) [reply yarl/__init__.py]: Sorry, don't follow the question.

File: yarl/__init__.py [hunk 2]
```diff
@@ -160,16 +160,16 @@ def __init__(self, val='', *, encoded=False, strict=None):
                 if netloc is None:
                     raise ValueError(
                         "Invalid URL: host is required for abolute urls.")
-                try:
-                    netloc.encode('ascii')
-                except UnicodeEncodeError:
-                    netloc = netloc.encode('idna').decode('ascii')
                 else:
+                    (ip, *zone) = netloc.split('%', 1)
```
Comment (asvetlov) [inline yarl/__init__.py]: use `.partition()`

**Comment by asvetlov** [comment]
Merged by 7d8f58b

**Pull request closed without merging**